core/src/test/java/io/confluent/kafka/schemaregistry/rest/RestApiModeTest.java

```diff
@@ -288,6 +288,65 @@ public void testRegisterSchemaWithSameIdAfterImport() throws Exception {
             restApp.restClient.getVersion(subject, 1).getSchema());
   }
 
+  @Test
+  public void testRegisterSchemaWithSameIdDifferentMetadataAfterImport() throws Exception {
+    String subject = "testSubject";
+    String mode = "READWRITE";
+
+    // set mode to read write
+    assertEquals(
+            mode,
+            restApp.restClient.setMode(mode).getMode());
+
+    int expectedIdSchema1 = 1;
+    assertEquals("Registering without id should succeed",
+            expectedIdSchema1,
+            restApp.restClient.registerSchema(SCHEMA_STRING, subject));
+
+    // delete subject so we can switch to import mode
+    restApp.restClient.deleteSubject(Collections.emptyMap(), subject);
+
+    mode = "IMPORT";
+
+    // set mode to import
+    assertEquals(
+            mode,
+            restApp.restClient.setMode(mode).getMode());
+
+    // register same schema with same id different metadata
+    expectedIdSchema1 = 1;
+    RegisterSchemaRequest request = new RegisterSchemaRequest();
+    request.setSchema(SCHEMA_STRING);
+    request.setMetadata(new Metadata(null, ImmutableMap.of("foo", "bar"), null));
+    request.setVersion(1);
+    request.setId(expectedIdSchema1);
+    assertEquals("Registering with id should succeed",
+        expectedIdSchema1,
+        restApp.restClient.registerSchema(request, subject, false).getId());
+
+    assertEquals("Getting schema by id should succeed",
+        SCHEMA_STRING,
+        restApp.restClient.getVersion(subject, 1).getSchema());
+
+    // delete subject again
+    restApp.restClient.deleteSubject(Collections.emptyMap(), subject);
+
+    // register same schema with same id different metadata
+    expectedIdSchema1 = 1;
+    request = new RegisterSchemaRequest();
+    request.setSchema(SCHEMA_STRING);
+    request.setMetadata(new Metadata(null, ImmutableMap.of("foo", "bar"), null));
```

Comment:
isn't this the same metadata as before?

Reply:
Yes it's the same metadata.  I changed the name of the test and the comments, it should have been "same id but with metadata"